## 문제 1. 스타벅스 서울 매장 위치 찾기

### 서울 시내 매장 위치 수집 페이지로 넘어가기

In [123]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By

url = "https://www.starbucks.co.kr/"
driver = webdriver.Chrome(service=Service("../driver/chromedriver"))
driver.set_window_size(1400, 1000)

driver.get(url)

In [124]:
store_banner = driver.find_element(By.CLASS_NAME, "gnb_nav03")
store_banner.click()

In [125]:
store_button = driver.find_element(By.CLASS_NAME, "store_bn1_btn")
store_button.click()

In [126]:
loca_btn = driver.find_element(By.CLASS_NAME, "loca_search")
loca_btn.click()

In [127]:
sido = driver.find_element(By.CLASS_NAME, "sido_arae_box")
sido_list = sido.find_elements(By.TAG_NAME, "li")

sido_names = [sido_name.text for sido_name in sido_list]
sido_names

['서울',
 '경기',
 '광주',
 '대구',
 '대전',
 '부산',
 '울산',
 '인천',
 '강원',
 '경남',
 '경북',
 '전남',
 '전북',
 '충남',
 '충북',
 '제주',
 '세종']

In [128]:
print(sido_names[0])
sido_list[0].click()

서울


In [129]:
gu = driver.find_element(By.CLASS_NAME, "gugun_arae_box")
gu_list = gu.find_elements(By.TAG_NAME, "li")

gu_names = [gu_name.get_attribute("innerText") for gu_name in gu_list]
print(len(gu_names), gu_names)

26 ['전체', '강남구', '강동구', '강북구', '강서구', '관악구', '광진구', '구로구', '금천구', '노원구', '도봉구', '동대문구', '동작구', '마포구', '서대문구', '서초구', '성동구', '성북구', '송파구', '양천구', '영등포구', '용산구', '은평구', '종로구', '중구', '중랑구']


In [72]:
print(gu_names[0])
gu_list[0].click()

전체


In [73]:
from bs4 import BeautifulSoup

page = driver.page_source
soup = BeautifulSoup(page, "html.parser")

star_list = soup.find(class_="quickSearchResultBoxSidoGugun")
print(star_list.prettify())

<ul class="quickSearchResultBoxSidoGugun">
 <li class="quickResultLstCon" data-code="3762" data-hlytag="null" data-index="0" data-lat="37.501087" data-long="127.043069" data-name="역삼아레나빌딩" data-storecd="1509" style="background:#fff">
  <strong data-my_siren_order_store_yn="N" data-name="역삼아레나빌딩" data-store="1509" data-yn="N">
   역삼아레나빌딩
  </strong>
  <p class="result_details">
   서울특별시 강남구 언주로 425 (역삼동)
   <br/>
   1522-3232
  </p>
  <i class="pin_general">
   리저브 매장 2번
  </i>
 </li>
 <li class="quickResultLstCon" data-code="3672" data-hlytag="null" data-index="1" data-lat="37.510178" data-long="127.022223" data-name="논현역사거리" data-storecd="1434" style="background:#fff">
  <strong data-my_siren_order_store_yn="N" data-name="논현역사거리" data-store="1434" data-yn="N">
   논현역사거리
  </strong>
  <p class="result_details">
   서울특별시 강남구 강남대로 538 (논현동)
   <br/>
   1522-3232
  </p>
  <i class="pin_general">
   리저브 매장 2번
  </i>
 </li>
 <li class="quickResultLstCon" data-code="3858" data-hlytag="null" 

In [74]:
import re

star_info = star_list.find_all("li")

name = star_info[0].find("strong").text
info = star_info[0].find("p").text

match = re.search(r'\d{4}-\d{4}', info)

if match:
    phone_number = match.group(0)  # 전화번호
    address = info.replace(phone_number, '').strip()  # 나머지는 주소
else:
    address = info  # 전화번호가 없을 경우 전체가 주소

print("이름:", name)
print("주소:", address)
print("전화번호:", phone_number)

이름: 역삼아레나빌딩  
주소: 서울특별시 강남구 언주로 425 (역삼동)
전화번호: 1522-3232


### 서울 전체에 적용, 데이터베이스에 저장

In [1]:
#DB 클래스 구축
import pandas as pd
import mysql.connector

class databaseManager:
    def __init__(self, host_name, port, user_name, pwd, db_name):
        self.host_name = host_name
        self.port = port
        self.user_name = user_name
        self.pwd = pwd
        self.db_name = db_name

    def getConnection(self):
        self.conn = mysql.connector.connect(
            host = self.host_name,
            port = self.port,
            user = self.user_name,
            password = self.pwd,
            database = self.db_name
        )

        self.cur = self.conn.cursor(buffered=True)

    def createTable(self, table_name, **kwargs):
        table_str = ""
        for key, value in kwargs.items():
            table_str = table_str + f"{key} {value}, "

        table_str = table_str.rstrip(', ')

        table_sql = f"CREATE TABLE IF NOT EXISTS {table_name} ({table_str})"
        self.cur.execute(table_sql)

    def addColumn(self, table_name, column_name, column_type):
        add_sql = f"ALTER TABLE {table_name} ADD COLUMN {column_name} {column_type}"
        self.cur.execute(add_sql)

    def setData(self, table_name, *args):
        value_str = ', '.join(['%s'] * len(args))
        sql = f"""INSERT INTO {table_name} VALUES ({value_str})"""

        self.cur.execute(sql, args)
        self.conn.commit()

    def setColumnData(self, table_name, column_name, value, find_name, find_value):
        col_str = f"UPDATE {table_name} SET {column_name}={value} WHERE {find_name}='{find_value}'"
        self.cur.execute(col_str)
        self.conn.commit()

    def getColumnData(self, table_name, column_name):
        self.cur.execute(f"SELECT {column_name} FROM {table_name}")
        result = self.cur.fetchall()

        result_list = [row[0] for row in result]
        return result_list

    def getData(self, table_name):
        self.cur.execute(f"select * from {table_name}")

        result = self.cur.fetchall()
        columns = [desc[0] for desc in self.cur.description]

        df = pd.DataFrame(result, columns=columns)
        return df
    
    def close(self):
        self.conn.close()

In [76]:
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.common.exceptions import StaleElementReferenceException
import time
from tqdm.notebook import tqdm

star_db = databaseManager(
    "kkachi-db-1.cdi866oq2142.ap-northeast-2.rds.amazonaws.com",
    3306,
    "root",
    "th0708csi!",
    "kkachi_db")

star_db.getConnection()
star_db.createTable("starbucks", name="varchar(32)", address="varchar(64)", phone_number="varchar(16)")

page = driver.page_source
soup = BeautifulSoup(page, "html.parser")
star_list = soup.find(class_="quickSearchResultBoxSidoGugun")

for star in tqdm(star_info):
    name = star.find("strong").text
    info = star.find("p").text

    match = re.search(r'\d{4}-\d{4}', info)

    if match:
        phone_number = match.group(0)  # 전화번호
        address = info.replace(phone_number, '').strip()  # 나머지는 주소
    else:
        address = info  # 전화번호가 없을 경우 전체가 주소

    star_db.setData("starbucks", name, address, phone_number)

star_df = star_db.getData("starbucks")
star_db.close()
star_df.tail(20)

  0%|          | 0/634 [00:00<?, ?it/s]

name                                    address phone_number
614      숙대입구역                     서울특별시 용산구 한강대로 291 (갈월동)    1522-3232
615  용산아이파크몰6F                         서울특별시 용산구 한강대로23길 55    1522-3232
616      동부이촌동                      서울특별시 용산구 이촌로 198 (이촌동)    1522-3232
617     숙명여대정문    서울특별시 용산구 청파로47길 100 (청파동2가) 프라임관B1F(로비층)    1522-3232
618   용산해링턴스퀘어                    서울특별시 용산구 서빙고로 17 (한강로3가)    1522-3232
619       경리단길                      서울특별시 용산구 회나무로 3 (이태원동)    1522-3232
620       서울역사              서울특별시 용산구 한강대로 405 (동자동) 신-101호    1522-3232
621     용산리첸시아           서울특별시 용산구 백범로 341 (원효로1가, 리첸시아 용산)    1522-3232
622     용산전자랜드                     서울특별시 용산구 청파로 74 (한강로3가)    1522-3232
623      용산이마트     서울특별시 용산구 한강대로23길 55 (한강로3가) 용산이마트 지하 2층    1522-3232
624  용산아이파크몰1F              서울특별시 용산구 한강대로23길 55 1층 (한강로3가)    1522-3232
625         상봉       서울특별시 중랑구 상봉로 131 (상봉동, 상봉 듀오트리스 주상복합)    1522-3232
626        중랑역                     서울특별시 중랑구 망우로30길 3 (상봉동)    1522-3232
627       중랑구청                             서울특별시 중랑구 신내로 72    1522-3232
628       사가정역                            서울특별시 중랑구 면목로 310    1522-3232
629        상봉역                      서울특별시 중랑구 망우로 307 (상봉동)    1522-3232
630         묵동        서울특별시 중랑구 동일로 952 (묵동, 로프트원 태릉입구역) 1층    1522-3232
631        망우동                      서울특별시 중랑구 망우로 407 (망우동)    1522-3232
632        양원역                    서울특별시 중랑구 양원역로10길 3 (망우동)    1522-3232
633        중화역                         서울특별시 중랑구 봉화산로 35 1층    1522-3232

In [77]:
driver.quit()

## 문제 2. 이디야커피 서울 매장 위치 찾기

### 서울 시내 매장 위치 수집 페이지로 넘어가기

In [141]:
url = "https://members.ediya.com/store"
driver = webdriver.Chrome(service=Service("../driver/chromedriver"))
driver.set_window_size(1400, 1000)

driver.get(url)

In [142]:
#del gu_names[0]
print(gu_names)

['강남구', '강동구', '강북구', '강서구', '관악구', '광진구', '구로구', '금천구', '노원구', '도봉구', '동대문구', '동작구', '마포구', '서대문구', '서초구', '성동구', '성북구', '송파구', '양천구', '영등포구', '용산구', '은평구', '종로구', '중구', '중랑구']


In [143]:
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys

search_tap = driver.find_element(By.CLASS_NAME, "srch_tab_wrap")
srch_btn = search_tap.find_elements(By.TAG_NAME, "a")[1]
srch_btn.click()

In [144]:
search_input = driver.find_element(By.TAG_NAME, "input")

search_input.send_keys(gu_names[0])
search_input.send_keys(Keys.ENTER)

### 매장 정보 수집 테스트

In [135]:
page = driver.page_source
soup = BeautifulSoup(page, "html.parser")

ediya_list = soup.find(class_="store_list")
print(ediya_list.prettify())

<div class="store_list" style="">
 <div class="st_li">
  <div class="info_txt">
   <h4 class="name">
    도산사거리점
   </h4>
   <p class="addr">
    서울특별시 강남구 언주로148길 14 나동 1층 107호
   </p>
  </div>
  <div class="img_area">
   <figure class="img">
    <img src="https://ediya-prod-membership.s3.ap-northeast-2.amazonaws.com/store/default/b68d0ffc-dae0-4d0c-b356-f37abebff2c8.jpeg"/>
   </figure>
  </div>
  <a class="go_view" href="javascript:storeDetailPopup('1601' , 'hide')">
   <span class="blind">
    매장선택
   </span>
  </a>
 </div>
 <div class="st_li">
  <div class="info_txt">
   <h4 class="name">
    학동제마점
   </h4>
   <p class="addr">
    서울 강남구 학동로34길 22 (논현동)
   </p>
  </div>
  <div class="img_area">
   <figure class="img">
    <img src="https://ediya-prod-membership.s3.ap-northeast-2.amazonaws.com/store/default/2386ef2c-10f3-4045-8f05-635c6bb0d5f9.jpeg"/>
   </figure>
  </div>
  <a class="go_view" href="javascript:storeDetailPopup('6217816' , 'hide')">
   <span class="blind">
    매장선택
 

In [136]:
ediya_info = ediya_list.find_all(class_="info_txt")

name = ediya_info[0].find("h4").text
full_addr = ediya_info[0].find("p").text
match = re.search(r'([가-힣]+(?:특별시|도) [가-힣]+(?:구) [가-힣0-9\s]+길\s*\d+)', full_addr)

if match:
    address = match.group(0).strip()  # 공백 제거 후 출력
else:
    print("주소를 찾을 수 없습니다.")

print("이름:", name)
print("주소:", address)

이름: 도산사거리점
주소: 서울특별시 강남구 언주로148길 14


### 서울 전체에 적용, 데이터베이스에 저장

In [145]:
ediya_db = databaseManager(
    "kkachi-db-1.cdi866oq2142.ap-northeast-2.rds.amazonaws.com",
    3306,
    "root",
    "th0708csi!",
    "kkachi_db")

ediya_db.getConnection()
ediya_db.createTable(
    "ediya", 
    name="varchar(32)", 
    address="varchar(48)"
)

for idx, gu in tqdm(enumerate(gu_names)):
    search_input = WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.TAG_NAME, "input")))

    search_input.clear()
    search_input.send_keys(gu)
    search_input.send_keys(Keys.ENTER)
    time.sleep(1)

    page = driver.page_source
    soup = BeautifulSoup(page, "html.parser")

    ediya_list = soup.find(class_="store_list")
    ediya_info = ediya_list.find_all(class_="info_txt")

    for info in tqdm(ediya_info):
        name = info.find("h4").text.strip()
        address = info.find("p").text
        if address[:2] == "서울" or address[:5] == "서울특별시":
            ediya_db.setData("ediya", name, address)
        else:
            continue

ediya_df = ediya_db.getData("ediya")
ediya_db.close()
ediya_df.tail(20)

0it [00:00, ?it/s]

  0%|          | 0/29 [00:00<?, ?it/s]

  0%|          | 0/22 [00:00<?, ?it/s]

  0%|          | 0/10 [00:00<?, ?it/s]

  0%|          | 0/29 [00:00<?, ?it/s]

  0%|          | 0/21 [00:00<?, ?it/s]

  0%|          | 0/17 [00:00<?, ?it/s]

  0%|          | 0/17 [00:00<?, ?it/s]

  0%|          | 0/15 [00:00<?, ?it/s]

  0%|          | 0/18 [00:00<?, ?it/s]

  0%|          | 0/15 [00:00<?, ?it/s]

  0%|          | 0/20 [00:00<?, ?it/s]

  0%|          | 0/17 [00:00<?, ?it/s]

  0%|          | 0/21 [00:00<?, ?it/s]

  0%|          | 0/19 [00:00<?, ?it/s]

  0%|          | 0/28 [00:00<?, ?it/s]

  0%|          | 0/16 [00:00<?, ?it/s]

  0%|          | 0/23 [00:00<?, ?it/s]

  0%|          | 0/24 [00:00<?, ?it/s]

  0%|          | 0/16 [00:00<?, ?it/s]

  0%|          | 0/35 [00:00<?, ?it/s]

  0%|          | 0/13 [00:00<?, ?it/s]

  0%|          | 0/20 [00:00<?, ?it/s]

  0%|          | 0/27 [00:00<?, ?it/s]

  0%|          | 0/50 [00:00<?, ?it/s]

  0%|          | 0/25 [00:00<?, ?it/s]

name                           address
489     서울의료원점                 서울 중랑구 신내로 156 1층
490    서울상봉1동점          서울 중랑구 상봉중앙로 36 (상봉동) 1층
491  서울중화초등학교점    서울 중랑구 신내로7가길 7 (신내동, 제이리옴) 1층
492       상봉역점     서울 중랑구 망우로 277 (상봉동) 1층 이디야커피
493  면목경남아너스빌점         서울 중랑구 면목로27길 92 (면목동) 1층
494       양원역점     서울 중랑구 양원역로14가길 24 (망우동) 101호
495      화랑대역점  서울 중랑구 신내로25가길 2 (묵동, 현동학당) 104호
496   사가정아이파크점   서울 중랑구 동일로92길 57 (면목동, 메디포레) 1층
497      봉화산역점         서울 중랑구 신내동  646 금강리빙스텔114
498       먹골역점         서울 중랑구 동일로157길 13 (묵동) 1층
499       중화동점         서울 중랑구 동일로129길 1 (중화동) 1층
500      서일중앙점       서울 중랑구 용마산로 394 1층, 1호(면목동)
501     중랑교차로점            서울 중랑구 동일로 683 1층(면목동)
502      금란망우점            서울 중랑구 망우로 460 1층(망우동)
503      상봉엠코점                서울특별시 중랑구 망우로 353 
504      상봉중앙점        서울특별시 중랑구 망우로 316  1층 103호
505      망우중앙점          서울특별시 중랑구 용마산로115길 109  
506       중랑역점                서울 중랑구 상봉동  130-99
507       신내동점                서울 중랑구 신내동  613-10
508      사가정역점                  서울 중랑구 면목로 325 .

In [146]:
driver.quit()

## 문제 3. 데이터 분석(이디야는 스타벅스를 따라오나?)

### 각 점포 좌표값 가져오기

In [2]:
db_1 = databaseManager(
    "kkachi-db-1.cdi866oq2142.ap-northeast-2.rds.amazonaws.com",
    3306,
    "root",
    "th0708csi!",
    "kkachi_db")

db_1.getConnection()

In [156]:
db_1.addColumn("starbucks", "lat", "float")
db_1.addColumn("starbucks", "lng", "float")

In [157]:
db_1.addColumn("ediya", "lat", "float")
db_1.addColumn("ediya", "lng", "float")

In [4]:
star_list = db_1.getColumnData("starbucks", "address")
ediya_list = db_1.getColumnData("ediya", "address")

len(star_list), len(ediya_list)

(634, 509)

In [180]:
import googlemaps

gmaps_key = "AIzaSyD2qjAAYOE_86HoatbAQ8Fn_Oc88e23gko"
gmaps = googlemaps.Client(key=gmaps_key)

for s_addr in tqdm(star_list):
    geocode = gmaps.geocode(address=s_addr)

    lat = geocode[0].get("geometry")["location"]["lat"]
    lng = geocode[0].get("geometry")["location"]["lng"]

    print(float(lat), float(lng))

    db_1.setColumnData("starbucks", "lat", float(lat), "address", s_addr)
    db_1.setColumnData("starbucks", "lng", float(lng), "address", s_addr)

  0%|          | 0/634 [00:00<?, ?it/s]

37.5000776 127.0385419
37.5136787 127.0317124
37.5136787 127.0317124
37.499615 127.0316758
37.4945248 127.062584
37.5150389 127.0631304
37.5228325 127.028577
37.5125207 127.0588194
37.5074573 127.0606353
37.52741049999999 127.0330945
37.4880028 127.1026639
37.4853306 127.0366142
37.5054927 127.0504299
37.5113321 127.0484374
37.5020948 127.0266668
37.5125207 127.0588194
37.5183507 127.0459641
37.493112 127.041268
37.5000776 127.0385419
37.49903460000001 127.0602191
37.5222132 127.0565181
37.5263249 127.0295435
37.5219144 127.0224932
37.5252727 127.0417742
37.5000776 127.0385419
37.5224015 127.0379633
37.5033843 127.0588842
37.5075307 127.0574271
37.5189305 127.0346042
37.5162574 127.0206194
37.4922077 127.0573813
37.5148904 127.0357573
37.5237869 127.0468574
37.4966281 127.055214
37.4942547 127.0328547
37.5125207 127.0588194
37.5232025 127.0216261
37.5018802 127.0392296
37.4994355 127.0335909
37.5030912 127.0255689
37.5243529 127.051588
37.5098065 127.0578853
37.5078813 127.0652928
37.5

In [7]:
from tqdm.notebook import tqdm

for e_addr in tqdm(ediya_list):
    geocode = gmaps.geocode(address=e_addr)

    lat = geocode[0].get("geometry")["location"]["lat"]
    lng = geocode[0].get("geometry")["location"]["lng"]

    print(float(lat), float(lng))

    db_1.setColumnData("ediya", "lat", float(lat), "address", e_addr)
    db_1.setColumnData("ediya", "lng", float(lng), "address", e_addr)

  0%|          | 0/509 [00:00<?, ?it/s]

37.5204069 127.0360954
37.51313870000001 127.03365
37.5004196 127.0630716
37.501434 127.052328
37.4735822 127.1078311
37.5038188 127.0481497
37.4979551 127.0561304
37.5200681 127.0491634
37.5165513 127.0401391
37.5150479 127.0331376
37.5140759 127.0280905
37.5222821 127.0314799
37.5194023 127.0294305
37.4958415 127.0405444
37.500366 127.042869
37.5104408 127.0650485
37.4942156 127.077617
37.4885331 127.0674474
37.4904095 127.0816397
37.478468 127.046625
37.5037879 127.0363296
37.5099434 127.0357023
37.524866 127.050373
37.5177943 127.0214406
37.4824238 127.0430395
37.5018112 127.0352111
37.4953995 127.030406
37.4985444 127.0479369
37.5068833 127.059437
37.5466824 127.1349233
37.5653789 127.1741146
37.5451714 127.1414108
37.5724114 127.1758706
37.5304813 127.1298584
37.55737130000001 127.1725399
37.5503165 127.1455692
37.5401616 127.1245175
37.5604113 127.1637115
37.5340436 127.1369525
37.5515869 127.1348633
37.5290236 127.1223472
37.5382414 127.1263464
37.5245419 127.1288923
37.547251 

### 좌표간 거리 기준으로 이디야와 가까운 스타벅스 찾기, 그리고 50m 기준으로 이디야 점포의 비율 찾기

In [16]:
from geopy.distance import geodesic

starbucks_data = db_1.getData("starbucks")
ediya_data = db_1.getData("ediya")

# 1. 가장 가까운 스타벅스까지의 거리 계산 함수
def compute_nearest_distance(store_df):
    results = []
    for _, store in store_df.iterrows():
        store_loc = (store['lat'], store['lng'])

        min_distance = float('inf')
        nearest_starbucks = None

        for _, starbucks in starbucks_data.iterrows():
            starbucks_loc = (starbucks['lat'], starbucks['lng'])
            distance = geodesic(store_loc, starbucks_loc).meters  # 거리(m)

            if distance < min_distance:
                min_distance = distance
                nearest_starbucks = starbucks['name']
                nearest_star_lat = starbucks['lat']
                nearest_star_lng = starbucks['lng']
        
        results.append([store['name'], store['lat'], store['lng'], nearest_starbucks, nearest_star_lat, nearest_star_lng, min_distance])

    return pd.DataFrame(results, columns=['매장명', '매장 위도', '매장 경도', '가까운 스타벅스', '근처 스벅 위도', '근처 스벅 경도', '거리 (m)'])

# 2. 이디야 점포와의 거리 계산
distance_df = compute_nearest_distance(ediya_data)

# 3. 거리 통계 계산
summary_stats = distance_df['거리 (m)'].agg(['mean', 'median'])
summary_stats.columns = ['평균 거리(m)', '중앙값 거리(m)']

# 4. 50m 이내 매장 비율 계산
threshold = 50
near_count = (distance_df['거리 (m)'] <= threshold).sum()
total_count = len(distance_df)
per_within_50m = (near_count/total_count) * 100

# 결과 합치기
summary_stats['50m 이내 비율(%)'] = per_within_50m

# 5. 결과 출력
print(distance_df)
print(total_count)
print(summary_stats)

        매장명    매장 위도    매장 경도   가까운 스타벅스  근처 스벅 위도  근처 스벅 경도      거리 (m)
0    도산사거리점  37.5204  127.036  을지병원사거리     37.5189   127.035  188.496475
1     학동제마점  37.5131  127.034   논현역사거리     37.5137   127.032  188.943383
2     삼성대치점  37.5004  127.063       대치     37.4997   127.062  117.706962
3     강남대치점  37.5014  127.052   선릉세화빌딩     37.5038   127.051  280.661654
4     강남율현점  37.4736  127.108     강남세곡     37.4673   127.100  994.844463
..      ...      ...      ...        ...       ...       ...         ...
504   상봉중앙점  37.5965  127.088      상봉역     37.5970   127.087  104.300070
505   망우중앙점  37.5977  127.094       상봉     37.5974   127.092  179.731721
506    중랑역점  37.5933  127.075      중랑역     37.5932   127.075   11.098879
507    신내동점  37.6077  127.097     중랑구청     37.6054   127.096  270.114887
508   사가정역점  37.5811  127.088     사가정역     37.5796   127.088  166.482826

[509 rows x 7 columns]
509
mean            416.873490
median          313.104386
50m 이내 비율(%)      7.072692
Name: 거리 (m), d

### 지도로 시각화하기

In [18]:
import folium

map = folium.Map(location=[37.5502, 126.982], zoom_start=11, tiles="OpenStreetMap")

for idx, row in distance_df.iterrows():
    if (row['거리 (m)'] <= threshold):
        folium.Marker([row['매장 위도'], row['매장 경도']], popup=f"이디야커피 {row['매장명']}").add_to(map)
        folium.Marker([row['근처 스벅 위도'], row['근처 스벅 경도']], popup=f"스타벅스 {row['가까운 스타벅스']}", icon=folium.Icon(color="green", icon="star")).add_to(map)
    else:
        continue

map

## 분석 결과
-----------------------------
* 본 프로그램에서는 이디야 커피의 서울시내 점포와 가까운 스타벅스 점포를, 50m 이내 기준으로 거리를 찾아 그 비율을 산정했다(50m 이내로 기준을 잡은 이유는 보통 그 정도 거리가 같은 건물 혹은 한 블럭 이내 거리이기 때문으로, 유사한 위치를 잡아내는데 유용할거라 판단했기 때문이다).
* 그 결과, 전체 509개의 이디야 점포 중 7.072692%가 근처 스타벅스 점포와 50m 이내, 즉 매우 근접한 거리에 있다.
* 한편 지도 시각화를 해봤을 때, 이디야의 각 점포와 근접한 스타벅스 점포의 경우, 주로 시내 중심가(종로, 중구) 혹은 상업, 업무, 주거지구(홍대, 여의도 등)에 몰려있는 것을 알 수 있다.
* 따라서, 서울 시내 전체로 봤을 때는 이디야가 꼭 스타벅스와 근접한 위치에 자리한다고 볼 순 없지만, 주요 구역에서는 이디야가 스타벅스와 근접한 위치를 특정하여 입점한다고 의심해볼 수 있다.